### ISSUE TITLE
How to exempt the `T2` rule?

### ISSUE THREAD

**Issue by masutaka** [looker-open-source/look-at-me-sideways#69]
Thanks for releasing v2! I am trying to introduce LAMS in CI.

First, I tried to exempt all the current errors, but I just couldn't exempt the `T2` rule.

## Workaround?

According to the source code, I need to exempt all T2~T10. When I exempted all of them, I was able to exempt the error as expected. Is this the intended behavior?

https://github.com/looker-open-source/look-at-me-sideways/blob/9450155d9dbc7e2b6156cf6b9024a2c4228586f9/rules/t2-10.js#L11-L31

manifest.lkml:

```
# LAMS
# rule_exemptions: {
#  T2:  "We may have to deal with this later."
#  T3:  "We may have to deal with this later."
#  T4:  "We may have to deal with this later."
#  T5:  "We may have to deal with this later."
#  T6:  "We may have to deal with this later."
#  T7:  "We may have to deal with this later."
#  T8:  "We may have to deal with this later."
#  T9:  "We may have to deal with this later."
#  T10: "We may have to deal with this later."
# }
```

## Exemption per error

If I exempt them in `manifest.lkml`, I don't notice that the number of errors increases afterward. Therefore, I tried the per-error exemption, but it did not work.

I think the reason is that `!exempt('')` always returns `true`. Is it possible to have a per-error exemption for `T2` rule?

https://github.com/looker-open-source/look-at-me-sideways/blob/9450155d9dbc7e2b6156cf6b9024a2c4228586f9/rules/t2-10.js#L129-L138

**Comment by fabio-looker** [comment]
Hi @masutaka - thanks for the error report, and sorry for just now getting to reviewing it.

I think the problem is just that I copied the "boilerplate" of the rule from a simpler single-rule file, and in this simpler case the whole file returns early when there is a global exemption (for speed and efficiency)... so the exempt checking function on line 45 doesn't consider to check the global exemptions because it assumed the file would have already returned before that (but this one doesn't because the file contains many rules and has to continue if any of them are not globally exempt)

If I've understood this correctly, this should be an easy fix I can do this coming week.

**Comment by fabio-looker** [comment]
Note to self: I should also review rules K1-4 and rule F3 for a similar issue

**Comment by acepeda2019** [comment]
@fabio-looker Love the linter! I'm in the process of adding this to our projects at my company, and I'm running into the same issue for K1-4. As a workaround, we are adding in placeholders with `pk1` in the name but would prefer to write the exemption.

```
dimension: pk1_placeholder {
  primary_key: yes
  hidden: yes
  }
```

**Comment by fabio-looker** [comment]
This is now published to NPM as version 2.1.0

Also, in what should be a major quality-of-life improvement for everyone, exempting rule T2 is now effectively exempting all of T2-T10, since T2 is just the umbrella rule that the subsequent rules provide more specifics on

**Comment by masutaka** [comment]
Thanks to you, I reduced the global rules in manifest.lkml to local rules for each .lkml file. I will reduce more exemptions in the future.
